staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher.go

```diff
@@ -67,13 +69,16 @@ func TestOnlySetFatalOnDecodeError(b bool) {
 }
 
 type watcher struct {
-	client        *clientv3.Client
-	codec         runtime.Codec
-	newFunc       func() runtime.Object
-	objectType    string
-	groupResource schema.GroupResource
-	versioner     storage.Versioner
-	transformer   value.Transformer
+	client         *clientv3.Client
+	codec          runtime.Codec
+	newFunc        func() runtime.Object
+	newListFunc    func() runtime.Object
+	objectType     string
+	resourcePrefix string
+	groupResource  schema.GroupResource
+	versioner      storage.Versioner
+	transformer    value.Transformer
+	storage        storage.Interface
```

Comment:
instead of introducing this circular dependency, I would actually really prefer to pass the
```
getCurrentRV func() [or func(ctx)?
```

callback

Reply:
yes, that is alternative, i was also considering something similar, ok, i'll update the pr.